test/StockQuoteAnalyzerTest.java

```diff
@@ -41,4 +64,145 @@ public void tearDown() throws Exception {
+
+    @Test (expectedExceptions = StockTickerConnectionError.class)
+    public void refreshShouldThrowStockTickerConnectionErrorWhenUnableToConnectToTickerSource() throws Exception {
+        analyzer = new StockQuoteAnalyzer("GM", generatorMock, audioMock);
+        when(generatorMock.getCurrentQuote()).thenThrow(new Exception());
+        analyzer.refresh();
+    }
+
+    @Test (expectedExceptions = InvalidAnalysisState.class)
+    public void getPercentChangeSinceCloseShouldThrowInvalidAnalysisStateWhenNoQuoteReceived() throws Exception {
+        analyzer = new StockQuoteAnalyzer("GM", generatorMock, audioMock);
+        analyzer.getPercentChangeSinceClose();
+    }
+
+    @Test (dataProvider = "QuoteClasses")
+    public void getPercentChangeSinceCloseShouldReturnPercentChangeWhenQuotesProvided(String symbol, String name, double previousClose, double lastTrade,  double change) throws Exception {
+        DecimalFormat formatter = new DecimalFormat("#######.##");
+        analyzer = new StockQuoteAnalyzer(symbol, generatorMock, audioMock);
+        when(generatorMock.getCurrentQuote()).thenReturn(new StockQuote(symbol, previousClose, lastTrade, change));
+        analyzer.refresh();
+        double expected = Double.parseDouble(formatter.format((change/previousClose)*100.0)); // Rounds to 2 decimal places
+        double result = analyzer.getPercentChangeSinceClose();
+
+        assertEquals(expected, result);
+        verify(generatorMock, times(1)).getCurrentQuote();
+    }
+
+    @Test (expectedExceptions = InvalidAnalysisState.class)
+    public void getPreviousCloseShouldThrowExceptionWhenWhenNoQuoteReceived() throws Exception {
+        analyzer = new StockQuoteAnalyzer("GM", generatorMock, audioMock);
+        when(generatorMock.getCurrentQuote()).thenReturn(null);
+        analyzer.refresh();
+        analyzer.getPreviousClose();
+    }
+
+    @Test (dataProvider = "QuoteClasses")
+    public void getPreviousCloseShouldReturnPreviousCloseWhenThereExistsACurrentQuote(String symbol, String name, double previousClose, double lastTrade,  double change) throws Exception {
+        analyzer = new StockQuoteAnalyzer(symbol, generatorMock, audioMock);
+        when(generatorMock.getCurrentQuote()).thenReturn(new StockQuote(symbol, previousClose, lastTrade, change));
+        analyzer.refresh();
+        assertEquals(analyzer.getPreviousClose(), previousClose);
+    }
+
+    @Test (dataProvider = "QuoteClasses")
+    public void playAppropriateAudioShouldCallAppropriateMethodWhenCalled(String symbol, String name, double previousClose, double lastTrade,  double change) throws Exception {
+        analyzer = new StockQuoteAnalyzer(symbol, generatorMock, audioMock);
+        when(generatorMock.getCurrentQuote()).thenReturn(new StockQuote(symbol, previousClose, lastTrade, change));
+
+        analyzer.refresh();
+        double result = analyzer.getPercentChangeSinceClose();
+        analyzer.playAppropriateAudio();
+        if(result >=1.0){
```

Comment:
When I start to have tests with this much logic, I consider rewriting them.  Why not have 3 different tests?